exec/java-exec/src/main/java/org/apache/drill/exec/server/options/FallbackOptionManager.java

```diff
@@ -102,6 +119,29 @@ public void setOption(OptionValue value) {
   }
 
   @Override
+  public void deleteOption(final String name, final OptionType type) {
+    try {
+      SystemOptionManager.getValidator(name); // ensure the option exists
+    } catch (final IllegalArgumentException e) {
+      throw UserException.validationError(e)
+        .build(logger);
+    }
+
+    // fallback if unable to delete locally
+    if (!deleteLocalOption(name, type)) {
```

Comment:
you should call `name.toLowerCase()` here to make sure implementation will always be case insensitive.